src/som/compiler/MixinBuilder.java

```diff
@@ -165,16 +175,20 @@ public MixinScope getInstanceScope() {
     return instanceScope;
   }
 
-  public MixinBuilder getOuterBuilder() {
-    return outerBuilder;
+  public MixinBuilder getOuterMixinBuilder() {
+    return outerMixinBuilder;
+  }
+
+  public MethodBuilder getOuterMethodBuilder() {
+    return outerMethodBuilder;
   }
```

Comment:
I can't think of any reasonable case where both may be required. As an extreme example, I thought of one case (I believe it's nonsense, but for the sake of thinking about the problem):

```
class A  = ( |x = 1.| )(
  public foo = (
   | y |
  y := 11.
    
    ^ objL()( public printXY = (x print y println.) )
  )
)

class B = ( | x  = 2.| )(
  public bar = (
    | y |
   y := 12.
     (A foo) printXY.
  )
)
```

What should be printed when bar is invoked? If the object literal closes over both the method and the outer class, then bar will print (1 and 11). If it closes or neither then it will print (2 and 12). If it closes over only the method then, in theory, it could print 2 (B.x) and 11 (A.foo.x). 

If the correct answer is some reason 2 and 11, then we might need to store a reference to both the enclosing method and the outer class (so that during lookup we can check the current activation first, then bubble out to the activation of the enclosing method, then back to the enclosing classes: bar->foo->B instead of bar->foo->A). 

Again, I think this example might just be nonsensical, so there is likely no need to ever have both.